test/calico/metadata_access_test.go

```diff
@@ -0,0 +1,139 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build packet aws aws_edge
+// +build e2e
+
+package calico_test
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/yaml"
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+)
+
+const (
+	// Define manifest as YAML and then unmarshal it to Go stuct, so it is easier to
+	// write and debug, as it can be copy-pasted to the YAML file and applied manually.
+	metadataAccessPodManifest = `
+apiVersion: v1
+kind: Pod
+metadata:
+  generateName: metadata-access-test-
+spec:
+  restartPolicy: Never
+  containers:
+  - image: alpine
+    name: foo
+    command:
+    - wget
+    args:
+    - -O
+    - /dev/null
+    - --timeout
+    - "5"
+`
+
+	metadataAccessNamespaceManifest = `
+apiVersion: v1
+kind: Namespace
+metadata:
+  generateName: metadata-access-test-
+`
+
+	retryInterval = 1 * time.Second
+	timeout       = 5 * time.Minute
+)
+
+func TestNoMetadataAccessRandomPod(t *testing.T) { //nolint:funlen
+	t.Parallel()
+
+	client := testutil.CreateKubeClient(t).CoreV1()
+	nsclient := client.Namespaces()
+
+	ns := &corev1.Namespace{}
+	if err := yaml.Unmarshal([]byte(metadataAccessNamespaceManifest), ns); err != nil {
+		t.Fatalf("failed unmarshaling namespace manifest: %v", err)
+	}
+
+	p := &corev1.Pod{}
+	if err := yaml.Unmarshal([]byte(metadataAccessPodManifest), p); err != nil {
+		t.Fatalf("failed unmarshaling pod manifest: %v", err)
+	}
+
+	ns, err := nsclient.Create(ns)
+	if err != nil {
+		t.Fatalf("failed creating Namespace: %v", err)
+	}
+
+	metadataAddress := ""
+	platform := os.Getenv("PLATFORM")
+
+	switch platform {
+	case testutil.PlatformPacket, testutil.PlatformPacketARM:
+		metadataAddress = "https://metadata.packet.net/metadata"
+	case testutil.PlatformAWS:
+		metadataAddress = "http://169.254.169.254/latest/meta-data"
+	}
+
+	if metadataAddress == "" {
+		t.Fatalf("Platform %q not supported", platform)
+	}
+
+	// Set the right address to query, which is platform-dependent.
+	p.Spec.Containers[0].Args = append(p.Spec.Containers[0].Args, metadataAddress)
+
+	podsclient := client.Pods(ns.ObjectMeta.Name)
+
+	p, err = podsclient.Create(p)
+	if err != nil {
+		t.Fatalf("failed creating Pod: %v", err)
+	}
+
+	phase := corev1.PodUnknown
+
+	if err := wait.PollImmediate(retryInterval, timeout, func() (done bool, err error) {
+		p, err := podsclient.Get(p.ObjectMeta.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		phase = p.Status.Phase
+
+		if phase == corev1.PodFailed || phase == corev1.PodSucceeded {
+			return true, nil
+		}
+
+		return false, nil
+	}); err != nil {
+		t.Errorf("error while waiting for the pod: %v", err)
+	}
+
+	if phase == corev1.PodSucceeded {
+		t.Fatalf("Randomly created pod should not have access to %q", metadataAddress)
```

Comment:
Nit: the use of randomly here is weird

```suggestion
		t.Fatalf("Pods should not have access to %q", metadataAddress)
```